src/js_engine/v8.rs

```diff
@@ -0,0 +1,180 @@
+            engine: Rc::clone(&self.0),
+        })
+    }
+
+    fn create_bool_value(&self, input: bool) -> Result<Self::JsValue<'_>> {
+        let engine: &mut V8Wrapper = &mut self.0.borrow_mut();
+        let scope = &mut engine.scope();
+        let local = v8::Local::from(v8::Boolean::new(scope, input));
+        Ok(Value {
+            value: v8::Global::new(scope, local),
+            engine: Rc::clone(&self.0),
+        })
+    }
+
+    fn create_int_value(&self, input: i32) -> Result<Self::JsValue<'_>> {
+        let engine: &mut V8Wrapper = &mut self.0.borrow_mut();
+        let scope = &mut engine.scope();
+        let local = v8::Local::from(v8::Integer::new(scope, input));
+        Ok(Value {
+            value: v8::Global::new(scope, local),
+            engine: Rc::clone(&self.0),
+        })
+    }
+
+    fn create_float_value(&self, input: f64) -> Result<Self::JsValue<'_>> {
+        let engine: &mut V8Wrapper = &mut self.0.borrow_mut();
+        let scope = &mut engine.scope();
+        let local = v8::Local::from(v8::Number::new(scope, input));
+        Ok(Value {
+            value: v8::Global::new(scope, local),
+            engine: Rc::clone(&self.0),
+        })
+    }
+
+    fn create_string_value(&self, input: String) -> Result<Self::JsValue<'_>> {
+        let engine: &mut V8Wrapper = &mut self.0.borrow_mut();
+        let scope = &mut engine.scope();
+        let local = v8::Local::from(v8::String::new(scope, &input).unwrap());
+        Ok(Value {
+            value: v8::Global::new(scope, local),
+            engine: Rc::clone(&self.0),
+        })
+    }
+
+    fn create_object_value<'a>(
+        &'a self,
+        input: impl Iterator<Item = (String, Self::JsValue<'a>)>,
+    ) -> Result<Self::JsValue<'a>> {
+        let input = input.collect::<Vec<_>>();
+        let engine: &mut V8Wrapper = &mut self.0.borrow_mut();
+        let scope = &mut engine.scope();
+        let object = v8::Object::new(scope);
+        for (k, v) in input {
+            let key = v8::String::new(scope, &k).unwrap();
+            let value = v8::Local::new(scope, v.value);
+            object.set(scope, key.into(), value);
+        }
+        let obj_handle = v8::Global::new(scope, v8::Local::from(object));
+
+        Ok(Value {
+            value: obj_handle,
+            engine: Rc::clone(&self.0),
+        })
+    }
+}
+
+pub struct Value {
+    value: v8::Global<v8::Value>,
+    engine: Rc<RefCell<V8Wrapper>>,
+}
+
+impl JsValue<'_> for Value {
+    fn into_string(self) -> Result<String> {
+        self.engine.borrow_mut().try_catch(|scope| {
+            let maybe = v8::Local::new(scope, self.value).to_string(scope).unwrap();
+            Ok(maybe.to_rust_string_lossy(scope))
+        })
+    }
+}
+
+struct V8Wrapper {
+    isolate: v8::OwnedIsolate,
+}
+
+impl V8Wrapper {
+    fn new() -> V8Wrapper {
+        INIT.call_once(|| {
+            let platform = v8::new_unprotected_default_platform(0, false).make_shared();
```

Comment:
In case you are wondering, you can read more about `new_unprotected_default_platform` in https://github.com/supabase/edge-runtime/pull/227, https://github.com/denoland/deno/pull/20495, and https://github.com/denoland/rusty_v8/issues/1381.

It would be nice to be able to call the "protected" platform initialization on the main thread, but it seems like there is no easy way to detect if code is running on the main thread, and we don't want to accidentally initialize on a thread other than the main thread and cause a memory error. We could alternatively expose a function that calls the "protected" platform initialization, for users to themselves call it on the main thread, and then have the unprotected initialization as a fallback in case users didn't call the initialization themselves.